core/src/main/java/com/rudderstack/android/sdk/core/RudderClient.java

```diff
@@ -409,8 +409,8 @@ public void alias(@NonNull String newId, @Nullable RudderOption option) {
         String prevUserId = null;
         if (traits.containsKey("userId")) {
             prevUserId = (String) traits.get("userId");
-        } else if (traits.containsKey("id")) {
-            prevUserId = (String) traits.get("id");
+        } else if (traits.containsKey("anonymousId")) {
```

Comment:
checking for the key `id` is also needed. `userId` can be found in any one of these keys under traits : `userId`, `id`.

Set prevUserId to anonymousId if it is null after looking for userId in traits

Reply:
done